ssl/s3_enc.c

```diff
@@ -155,6 +155,7 @@ int ssl3_change_cipher_state(SSL *s, int which)
         RECORD_LAYER_reset_read_sequence(&s->rlayer);
         mac_secret = &(s->s3->read_mac_secret[0]);
     } else {
+        s->statem.invalid_enc_write_ctx = 1;
         if (s->enc_write_ctx != NULL) {
             reuse_dd = 1;
```

Comment:
Do we actually need to claim that the write_ctx is invalid if we are in the ```reuse_dd``` case?

Reply:
Yes, because EVP_CIPHER_CTX_reset resets ctx->cipher to Zero
and a lot of EVP_CIPHER functions like EVP_CIPHER_CTX_iv_length
dereference that member without further checks.

Reply:
It is luck if there is only a crash, it could be worse if the data is sent unencryped.